pkg/pipelines/tekton/pipelines_integration_test.go

```diff
@@ -114,35 +113,6 @@ func (p pl) log(args ...any) {
 	fmt.Fprintln(os.Stderr, args...)
 }
 
-func (p pl) SetTotal(i int) {
-	p.log("ProgressListener::SetTotal: ", i)
-}
-
-func (p pl) Increment(message string) {
-	p.log("ProgressListener::Increment: ", message)
-	if strings.Contains(message, "URL:") {
-		parts := strings.Split(message, "URL:")
-		if len(parts) < 2 {
-			p.log("bad output message: %q", message)
-			return
-		}
-		u := strings.TrimSpace(parts[1])
-		p.urlChan <- u
```

Comment:
Removal of this causes timeout of test since it never gets message that function has been deployed. The are additional changes required to make test work again.

Reply:
For instance we could add `io.Reader`'s for error/normal outputs into the `PipelinesProvider` structure and print to them instead to `os.Std[out|err]`. The test then could pass pipe when constructing `PipelinesProvider` and inspect output for occurrence of `exposed at URL:\n(\w)`.

wdyt @lkingland

Reply:
@gauron99 other maybe cleaner way would be to also return `fn.Instance` from the `Run()` method.

Reply:
The point is that `Run()` should somehow return url of deployed function. Be it via return value or output.

Reply:
So you can try invoking it.